server/pulp/server/webservices/views/search.py

```diff
@@ -64,7 +64,8 @@ def _parse_args(cls, args):
         options = {}
         for field in cls.optional_fields:
             if search_params.get(field):
-                options[field] = search_params.pop(field)
+                options[field] = args.get(field)
```

Comment:
perhaps:

    if isinstance(args.get(field), basestring):
        options[field] = args.get(field).lower == 'true'
    else:
         options[field] = args.get(field)
    search_params.pop(field)


This will require some changes in the subclasses to make sure that they expect bools, but it should fix the problem more generally.